config/routes.rb

```diff
@@ -1,3 +1,19 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  namespace :api do
+    namespace :v1 do
+      post '/auth/sign_in', to: 'authentication#login'
+      post '/auth', to: 'users#create'
+
+      resources :projects do
+        resources :tasks, except: %i[show] do
```

Comment:
please, use `shallow: true`.

It will cut long urls like this:
`/api/v1/projects/1/tasks/1`
to this:
`/api/v1/tasks/1`

Because task with id = 1 belongs to only one project, so we do not need to see `/projects/1/` in url